### ISSUE TITLE
When fails to auto-resolve attribute from http request data, should return 400 instead of 500

### ISSUE THREAD

**Issue by Y-Sindo** [Azure/azure-webjobs-sdk#2713]
#### Repro steps

1. Use this link to deploy to Azure. https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FY-Sindo%2Ffn-http-queue-s%2Fmaster%2Fazuredeploy.json

2. Call the function api without query parameters. For example, in Azure portal, click the function resource, "Functions" - "http-queue" - "Code + Test" - "Test/Run"

#### Expected behavior

400 error indicates that a query parameter "queue" not found. 

#### Actual behavior

500 error. 

#### Related information 

* Links to source: https://github.com/Y-Sindo/fn-http-queue-s

**Comment by v-anvari** [comment]
Hi @Y-Sindo , I am unable to repro it with the source you gave. Can you give us more information regarding the repro steps.

**Comment by Y-Sindo** [comment]
Here is the snapshots of the test step:
![image](https://user-images.githubusercontent.com/15143836/118751158-69395380-b893-11eb-956a-e6496c05336d.png)
![image](https://user-images.githubusercontent.com/15143836/118751179-72c2bb80-b893-11eb-91bc-7b8b919d9c31.png)

**Comment by v-anvari** [comment]
Hi @Y-Sindo , I have tried to reproduce this issue further and checked the parameters in your function.json file, and found that the queue name is given as below:
![image](https://user-images.githubusercontent.com/66122429/119019245-a1878180-b9ba-11eb-9ccc-4f31a7f26f6d.png)

The correct way of configuring the queue name is mentioned below, which is explained in the document [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-expressions-patterns#binding-expressions---app-settings):

{
  "bindings": [
    {
      "name": "order",
      "type": "queueTrigger",
      "direction": "out",
      "queueName": "%input_queue_name%",
      "connection": "MY_STORAGE_ACCT_APP_SETTING"
    }
  ]
}

Let us know if this helps.

**Comment by Y-Sindo** [comment]
No. Your example is a trigger while mine is just an input binding. In the queue input binding,  the queue name can be in the form `{....}`  or `%AppSetting%` , because `queueName` is an [AutoResolve] attribute.

**Comment by Y-Sindo** [comment]
We got an issue [Probable DDoS vulnerability](https://github.com/Azure/azure-functions-signalrservice-extension/issues/199) in SignalR Functions extensions repos. We also depend on the [AttributeCloner](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs) to resolve the `[AutoResolve]` attribute, so SignalR extensions behave similarly as queue input binding when fail to resolve an attribute from HTTP request. I think this is a common issue and should be better to be solved in WebJob SDK.

**Comment by v-anvari** [comment]
Hi @Y-Sindo , The function.json shows direction out which says its an output binding. I was mentioning the queue name in the output binding from your source

**Comment by Y-Sindo** [comment]
Yes, that's a mistake. Anyway, the queue should be allowed to be `{query.name}` by design.

**Comment by v-anvari** [comment]
Thank you for your feedback! We will investigate this further and update as appropriate.

**Comment by v-anvari** [comment]
Flagging this issue as feature request as it requires a validation for considering it in the breaking change
